mkosi/__init__.py

```diff
@@ -341,7 +341,12 @@ def finalize_scripts(scripts: Mapping[str, Sequence[PathString]] = {}) -> Iterat
                     DIR="$(cd "$(dirname "$0")" && pwd)"
                     PATH="$(echo "$PATH" | tr ':' '\\n' | grep -v "$DIR" | tr '\\n' ':')"
                     export PATH
-                    exec {shlex.join(str(s) for s in script)} "$@"
+                    if [ $# -gt 0 ]; then
+                        args="$@"
+                    else
+                        args=("sh" "-i")
+                    fi
+                    exec {shlex.join(str(s) for s in script)} "$args"
```

Comment:
POSIX shell doesn't have arrays, i.e. it's a bashism or am I misremembering that? At least in dash, which will be `/bin/sh` on Debian, 
```shell
args=("sh" "-i")
```
gives me a syntax array.

I think one could
```suggestion
                        args='sh - i'
                    fi
                    exec {shlex.join(str(s) for s in script)} $args
```
but this will lead to to globbing/expansion on the exec line, I think.

Reply:
I replaced this by just copying the `exec` line into both branches. It's maybe not the most sophisticated solution, but it should suffice ;)

Reply:
That's good enough. :)